Find dat folder.

kozlac opened the pull request:
Made changes to be able to find the dat folder when running from out of source folder. This resolves issue #604

intendo commented:
@sobolevnrm I approve the changes made to find the .DAT files (e.g. PARSE.DAT).
Do you want to review the changes and approve the pull request?

sobolevnrm commented:
@intendo - I reviewed when it showed up in my queue and posted some follow-on questions for the underlying issue #604.  I'm not trying to be difficult but I want to make sure we're "fixing" the right problem.  It's not obvious that this patch is preferable to developing a different method for managing paths in PDB2PQR -- I'd like to make sure we're considering larger changes that will make the code more useful/sustainable in the long run.

kozlac commented:
Added some new changes so that we can find the /dat folder regardless of where pdb2pqr is being run. Also, I added the setup.py script so that we can run pdb2pqr as a module. I'm excluding pdb2pka from the installation for now. And now that I remember I may be missing some dependencies in the setup script. 

The building and installation are the same as propka:
`$ cd pdb2qqr`
`$ python3 -m pip install --upgrade .`
`$ python3 -m pdb2pqr  [flags|options] <file_name>.pdb <file_name>.pqr`

sobolevnrm commented:
I will try to review this weekend; sorry for the delay.

sobolevnrm commented on pdb2pqr/pdb2pqr/main.py:
```diff
@@ -431,9 +435,12 @@ def non_trivial(args, protein, definition, is_cif):
+
+            protein.apply_pka_values(forcefield_.name, args.ph, dict(zip(pka_df.group_label, pka_df.pKa)));
+
+            # _LOGGER.error("\n" + str(pka_df))
```
Can you remove these commented-out lines, please?

sobolevnrm replied:
What happened to following the PEP8 style guide?  I just looked at this code and it has both a semicolon and an overly long line.

sobolevnrm replied:
I fixed this with https://github.com/Electrostatics/apbs-pdb2pqr/pull/605/commits/0ee632b7c49e86276053dce64a8c3fde42c5d8c7

sobolevnrm commented on pdb2pqr/setup.py:
```diff
@@ -0,0 +1,57 @@
+    name = "pdb2pqr",
+    version = VERSION,
+    author = "Baker, Nathan A. et al.",
+    author_email = "nathan.baker@pnnl.gov",
```
Let's use nathanandrewbaker@gmail.com here.

sobolevnrm replied:
This code wasn't de-linted either....

sobolevnrm replied:
Fixed in https://github.com/Electrostatics/apbs-pdb2pqr/pull/605/commits/fd4aafc8d390f46f8c49618e95ba0b07364eb4ad

sobolevnrm commented on pdb2pqr/setup.py:
```diff
@@ -0,0 +1,57 @@
+    long_description = long_description,
+    install_requires = ["propka >= 3.2", "pandas >= 1.0", "pytest>=5.4.1"],
+    url = " https://github.com/Electrostatics/apbs-pdb2pqr/tree/master/pdb2pqr",
+    packages = setuptools.find_packages(exclude = ["pdb2pka", "*.pdb2pka", "pdb2pka.*", "*.pdb2pka.*"]),
```
I don't think we should include `pdb2pka` yet -- it's not functional.  Is it needed for this PROPKA-only fix?

sobolevnrm replied:
Oops -- I read this wrong; it's excluding pdb2pka. Thanks!

sobolevnrm commented on pdb2pqr/setup.py:
```diff
@@ -0,0 +1,57 @@
+    },
+    license = "BSD",
+    classifiers = [
+        "Programming Language :: Python :: 3",
```
We probably want to include 3.5-3.8 explicity since we don't support versions <3.5

sobolevnrm replied:
Fixed in https://github.com/Electrostatics/apbs-pdb2pqr/pull/605/commits/fd4aafc8d390f46f8c49618e95ba0b07364eb4ad

sobolevnrm commented:
@kozlac and @intendo --  do we have test coverage for this PR?

sobolevnrm commented:
> Added some new changes so that we can find the /dat folder regardless of where pdb2pqr is being run. Also, I added the setup.py script so that we can run pdb2pqr as a module. I'm excluding pdb2pka from the installation for now. And now that I remember I may be missing some dependencies in the setup script.
> 
> The building and installation are the same as propka:
> `$ cd pdb2qqr`
> `$ python3 -m pip install --upgrade .`
> `$ python3 -m pdb2pqr [flags|options] <file_name>.pdb <file_name>.pqr`

Is this documented anywhere?  If not, can we add the documentation?

Note that for development, we probably want to suggest `pip install -e .`.
